### DISCUSSION TITLE
Why "Settings" localization is not included in package

### DISCUSSION THREAD

**Discussion by LaurentGrenet** [Open-Shell/Open-Shell-Menu#1627] [General]
OpenShell is translated in several languages... except its "Settings" dialog, that is only in english.

Nevertheless, it is possible to download some very old dll language files, that come from ClassicShell.
Obviously, these dll are not maintained, and are more and more obsolete with evolutions of OpenShell.

It seems to me that it would be a lot simpler to manage, and would allow to make it evolutive, to add these localizations in the standard package (even if not available for all languages)

Is there any chance to have these localizations included in a future version of OpenShell ?

**Comment by ge0rdi** [comment]
> OpenShell is translated in several languages... except its "Settings" dialog, that is only in english.

Things that `Open-Shell` draws/displays by itself (like start menu) use localizations from `*.L10N.ini` files that are included in installation.
These were inherited from original `Classic-Shell` and are easy to fix/adjust (anyone can submit PR that fixes some strings in them).

System dialogs cannot be localized this way (well, they possibly can but it would be much more complicated).
For those there are these language DLLs (you mentioned) that contain copies of original english resources that were then manually replaced by given language localizations using tools like ResourceHacker.

We are simply not maintaining those DLLs. Mostly because of that weird way they are supposed to be modified (that not much people can do).
That is also the main reason they aren't included in install package (as they are obsolete/incomplete), so we are rather providing link to those DLLs (on main page), so that anyone interested could use at least those old files.

> Is there any chance to have these localizations included in a future version of OpenShell ?

It would be nice if _somebody_ could maintain those DLLs.
In ideal case we should rather build them during build process (instead of manually hacking).
That will require to extract strings from existing files, create VS project for each one and put those strings (and other data) to `.rc` file so they will be in textual form that is easier to maintain.

So far there was nobody who could be able to do that.
